Rubberduck.Core/UI/CodeExplorer/Commands/UpdateFromFileCommand.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Rubberduck.Interaction;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Parsing.VBA.Extensions;
+using Rubberduck.Resources;
+using Rubberduck.VBEditor;
+using Rubberduck.VBEditor.Events;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.Extensions;
+using Rubberduck.VBEditor.SafeComWrappers;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using Rubberduck.VBEditor.Utility;
+
+namespace Rubberduck.UI.CodeExplorer.Commands
+{
+    public class UpdateFromFilesCommand : ImportCommand
+    {
+        private readonly IDeclarationFinderProvider _declarationFinderProvider;
+        private readonly IProjectsProvider _projectsProvider;
+        private readonly IModuleNameFromFileExtractor _moduleNameFromFileExtractor;
+
+        public UpdateFromFilesCommand(
+            IVBE vbe,
+            IFileSystemBrowserFactory dialogFactory,
+            IVbeEvents vbeEvents,
+            IParseManager parseManager,
+            IDeclarationFinderProvider declarationFinderProvider,
+            IProjectsProvider projectsProvider,
+            IModuleNameFromFileExtractor moduleNameFromFileExtractor,
+            IMessageBox messageBox)
+            : base(vbe, dialogFactory, vbeEvents, parseManager, messageBox)
+        {
+            _projectsProvider = projectsProvider;
+            _declarationFinderProvider = declarationFinderProvider;
+            _moduleNameFromFileExtractor = moduleNameFromFileExtractor;
+        }
+
+        protected override string DialogsTitle => RubberduckUI.UpdateFromFilesCommand_DialogCaption;
+
+        protected override void ImportFiles(ICollection<string> filesToImport, IVBProject targetProject)
+        {
+            var finder = _declarationFinderProvider.DeclarationFinder;
+
+            var moduleNames = ModuleNames(filesToImport);
+
+            var formBinaryModuleNames = moduleNames
+                .Where(kvp => ComponentTypeExtensions.FormBinaryExtension.Equals(Path.GetExtension(kvp.Key)))
+                .Select(kvp => kvp.Value)
+                .ToHashSet();
+
+            var formFilesWithoutBinaries = FormFilesWithoutBinaries(moduleNames, formBinaryModuleNames);
+
+            //We cannot import the the binary separately.
+            foreach (var formBinaryModuleName in formBinaryModuleNames)
+            {
+                moduleNames.Remove(formBinaryModuleName);
+            }
+
+            if (!ValuesAreUnique(moduleNames))
+            {
+                NotifyUserAboutAbortDueToDuplicateComponent(moduleNames);
```

Comment:
I think it would make for a better UX if the errors were all bunched up and reported once. So instead of having an user import, failing due to duplicate selection, removes the duplicate, then fails again due to non-matching file extensions, fixes up the file extensions, then fail again due to missing binaries files. That's lot of attempts.

Instead, the user should be presented with a message similar to this:

```
We found the following issue(s) with the selected files for import:

Duplicate files selected:
  - 'bar' matched by 'foo/bar.cls', 'baz/bar.cls'
  - 'baz' matched by 'foo/baz.cls', 'baz/baz.cls', 'bar/bar.cls'

Userform file missing its binary file:
  - 'userform1.frm'

Unrecognized file extension:
 - 'bar.bar'
 - 'baz.baz'
```

which would allow the user to correct multiple issues in one go and thus understand faster what the system expect the user to do.

Reply:
I do not intend to change the error reporting in this PR any further.

THB, I do not think that the different checks will trigger too often. E.g. to get two files for the same module, you have to actually rename one from the default export name. So, I do not really think that a joined reporting of all errors is really worth the effort.